charts/opentelemetry-collector/templates/_helpers.tpl

```diff
@@ -111,54 +111,27 @@ Return if ingress is stable.
   {{- eq (include "ingress.apiVersion" .) "networking.k8s.io/v1" -}}
 {{- end -}}
 
-
-{{- define "opentelemetry-collector.daemonset-podAnnotations" -}}
-{{- if eq .Values.mode "daemonset" }}
-  {{- if .Values.podAnnotations }}
-  {{- .Values.podAnnotations | toYaml }}
-  {{- end }}
-{{- else if .Values.agentCollector.podAnnotations }}
-{{- .Values.agentCollector.podAnnotations | toYaml }}
-{{- end }}
-{{- end }}
-
-{{- define "opentelemetry-collector.deployment-podAnnotations" -}}
-{{- if eq .Values.mode "deployment" }}
-  {{- if .Values.podAnnotations }}
-  {{- .Values.podAnnotations | toYaml }}
-  {{- end }}
-{{- else if .Values.standaloneCollector.podAnnotations }}
+{{- define "opentelemetry-collector.podAnnotations" -}}
+{{- if and .Values.podAnnotations .Values.mode  }}
+{{- .Values.podAnnotations | toYaml }}
+{{- else if and .Values.standaloneCollector.enabled .Values.standaloneCollector.podAnnotations }}
 {{- .Values.standaloneCollector.podAnnotations | toYaml }}
 {{- end }}
 {{- end }}
 
-{{- define "opentelemetry-collector.daemonset-podLabels" -}}
-{{- if eq .Values.mode "daemonset" }}
-  {{- if .Values.podLabels }}
+{{- define "opentelemetry-collector.podLabels" -}}
+{{- if and .Values.podLabels .Values.mode }}
```

Comment:
This change breaks backwards compatibility with .Values.agentCollector.podAnnotations

Reply:
The consolidation is still off.  With the removal of

```
{{- define "opentelemetry-collector.deployment-podAnnotations" -}}
{{- if eq .Values.mode "deployment" }}
  {{- if .Values.podAnnotations }}
  {{- .Values.podAnnotations | toYaml }}
  {{- end }}
{{- else if .Values.standaloneCollector.podAnnotations }}
{{- .Values.standaloneCollector.podAnnotations | toYaml }}
{{- end }}
{{- end }}
```

I don't see where users of .Values.standaloneCollector.podAnnotations would have their podAnnotations set.

I will be submitting a PR today to remove agentCollector and standaloneCollector.  Once that is out of the way adding pod annotations for daemonsets shouldn't need to worry about consolidation anymore.